### ISSUE TITLE
Certificate access denied

### ISSUE THREAD

**Issue by Abdruggi** [Azure/service-fabric-issues#423]
According to the documentation (https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-securing-containers) it should be possible to import the provided certificate inside a container (C# example code provided in the documentation). Unfortunately, the application process running as ContainerAdministrator has no access to the key and cert file. Just the System user is able to read it. A workaround would be highly appreciated.

**Comment by Abdruggi** [comment]
Got it. It is necessary to grant access to the file for the current user before reading the file

```c#
var acl = File.GetAccessControl(filePath);
acl.AddAccessRule(new FileSystemAccessRule(Environment.UserName, FileSystemRights.Read, AccessControlType.Allow));
File.SetAccessControl(filePath, acl);
```

**Comment by Abdruggi** [comment]
It would probably be a good idea to mention that in the documentation.

**Comment by mani-ramaswamy** [comment]
This issue is fixed in 6.1, so the container admin will have access. Closing this issue - please reopen as appropriate.

**Comment by MarkDixonTech** [comment]
This is an issue again when using microsoft/dotnet:2.2-aspnetcore-runtime-nanoserver-1803 since the user is set to ContainerUser which is then used as to run the entry point and does not have permission to read the certificate or password. 

The workaround is to add USER ContainerAdministrator just before the entry point
